pkg/transport/generic_transport_controller.go

```diff
@@ -403,65 +417,123 @@ func (c *genericTransportController) getObjectsFromWDS(ctx context.Context, plac
 	return objectsToPropagate, nil
 }
 
+func (c *genericTransportController) initializeWrappedObject(ctx context.Context, placementDecision *v1alpha1.PlacementDecision) (*unstructured.Unstructured, error) {
+	objectsToPropagate, err := c.getObjectsFromWDS(ctx, placementDecision)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get objects to propagate to WECs from PlacementDecision object '%s' - %w", placementDecision.GetName(), err)
+	}
+
+	wrappedObject, err := convertObjectToUnstructured(c.transport.WrapObjects(objectsToPropagate))
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert wrapped object to unstructured - %w", err)
+	}
+	// wrapped object name is (PlacementDecision.GetName()-WdsName).
+	// pay attention - we cannot use the PlacementDecision object name, cause we might have duplicate names coming from different WDS spaces.
+	// we add WdsName to the object name to assure name uniqueness,
+	// in order to easily get the origin PlacementDecision object name and wds, we add it as an annotations.
+	wrappedObject.SetName(fmt.Sprintf("%s-%s", placementDecision.GetName(), c.wdsName))
+	setLabel(wrappedObject, originOwnerReferenceLabel, placementDecision.GetName())
+	setLabel(wrappedObject, originWdsLabel, c.wdsName)
+	setAnnotation(wrappedObject, originOwnerGenerationAnnotation, placementDecision.GetGeneration())
+
+	return wrappedObject, nil
+}
+
+// pops wrapped object by namespace from the list and returns the requested wrapped object.
+// if the object is not found, list remains the same and nil is returned.
+// since the order of items in the list is not important, the implementation is efficient and was done as follows:
+// the functions goes over the list, if the requested object is found, it's replaced with the last object in the list.
+// then the function removes the last object in the list and returns the object.
+// in worst case where object is not found, it will go over all items in the list.
+func (c *genericTransportController) popWrappedObjectByNamespace(list *unstructured.UnstructuredList, namespace string) *unstructured.Unstructured {
+	found := false
+	length := len(list.Items)
+	for i := 0; i < length; i++ {
+		if list.Items[i].GetNamespace() == namespace {
+			currentObject := list.Items[i]
+			list.Items[i] = list.Items[length-1]
+			list.Items[length-1] = currentObject
+			found = true
+			break
+		}
+	}
+	if !found {
+		return nil
+	}
+	// otherwise, object was found and was replaced with last object in the list
+	poppedObject := &list.Items[length-1]
+	list.Items = list.Items[:length-1]
+	return poppedObject
```

Comment:
The control flow would be simpler if these statements were moved to replace the `break` above. And that would make some more simplification obvious (no need to put the returned object at the end of the list).

Reply:
fixed